tests/pipeline/test_blaze.py

```diff
@@ -1483,6 +1483,112 @@ def test_deltas_before_index_0(self, asset_info, add_extra_sid):
+        # has an earlier timestamp but later asof date so it should be
+        # selected.
+        #
+        # The third and fourth rows both map to index 1 in the output. The
+        # fourth row (second in the group) has both a later timestamp and asof
+        # date so it should be selected.
+        #
+        # The fifth and sixth rows both map to index 2 in the output. The fifth
+        # row (first in the group) has an earlier timestamp but later asof date
+        # so it should be selected.
+        deltas_df_single_sid = pd.DataFrame({
+            'value': [
+                0.0,  # selected
+                1.0,  # ignored
+
+                2.0,  # ignored
+                3.0,  # selected
+
+                4.0,  # selected
+                5.0,  # ignored
+            ],
+            'asof_date': [
+                # swapped order: second row is before the first
+                T('2014-01-02'),
+                T('2014-01-01'),
+
+                # chronological order: second row is after the first
+                T('2014-01-03'),
+                T('2014-01-04'),
+
+                # swapped order: second row is before the first
+                T('2014-01-06'),
+                T('2014-01-05'),
+            ],
+            'timestamp': [
+                # we learn about all rows in monotonically increasing order
+                T('2013-01-02 22:00'),
+                T('2014-01-02 23:00'),
+                T('2014-01-04 22:00'),
+                T('2014-01-04 23:00'),
+                T('2014-01-06 22:00'),
+                T('2014-01-06 23:00'),
+            ],
+        })
+        sids = asset_info.index
+        if add_extra_sid:
+            # add a sid to the dataset that the asset finder doesn't know about
+            sids = sids.insert(0, ord('Z'))
+
+        deltas_df = pd.concat([
+            deltas_df_single_sid.assign(
+                sid=sid,
+                value=deltas_df_single_sid.value + (100 * n),
+            )
+            for n, sid in enumerate(asset_info.index)
+        ])
+        deltas = bz.data(deltas_df, name='deltas', dshape=self.dshape)
+
+        expected_views_single_sid = keymap(pd.Timestamp, {
+            '2014-01-05': np.array([[0.0],
+                                    [3.0]]),
+            '2014-01-07': np.array([[3.0],
+                                    [4.0]]),
+        })
+
+        column_constant = np.arange(len(asset_info)) * 100
+        expected_views = {
+            k: v + column_constant
+            for k, v in expected_views_single_sid.items()
+        }
+        with tmp_asset_finder(equities=asset_info) as finder:
+            # The dates queried are non-contiguous. We have two day groups to
+            # capture the two day pairs in the input data.
+            dates = pd.to_datetime(['2014-01-03', '2014-01-05', '2014-01-07'])
+            self._run_pipeline(
+                expr,
+                deltas,
+                None,
```

Comment:
What are the `None`s here?  Might help to either pass these by keyword or comment inline.

Reply:
it's expected output, which can now be None if `compute_fn` is None to just check the views but not the full output. This is what you suggested in the other PR. I will pass these by kwarg.